Makefile.common

```diff
@@ -173,17 +195,28 @@ common-tarball: promu
 	@echo ">> building release tarball"
 	$(PROMU) tarball --prefix $(PREFIX) $(BIN_DIR)
 
-.PHONY: common-docker
-common-docker:
-	docker build -t "$(DOCKER_REPO)/$(DOCKER_IMAGE_NAME):$(DOCKER_IMAGE_TAG)" .
-
-.PHONY: common-docker-publish
-common-docker-publish:
-	docker push "$(DOCKER_REPO)/$(DOCKER_IMAGE_NAME)"
-
-.PHONY: common-docker-tag-latest
-common-docker-tag-latest:
-	docker tag "$(DOCKER_REPO)/$(DOCKER_IMAGE_NAME):$(DOCKER_IMAGE_TAG)" "$(DOCKER_REPO)/$(DOCKER_IMAGE_NAME):latest"
+.PHONY: common-docker $(BUILD_DOCKER_ARCHS)
+common-docker: $(BUILD_DOCKER_ARCHS)
+$(BUILD_DOCKER_ARCHS): common-docker-%:
+	docker build -t "$(DOCKER_REPO)/$(DOCKER_$*_IMAGE_NAME):$(DOCKER_IMAGE_TAG)" \
+		--build-arg BASE_IMAGE="$(DOCKER_$*_BASE_IMAGE)" \
+		--build-arg BIN_PATH="$(DOCKER_$*_BIN_PATH)" \
+		.
+
+.PHONY: common-docker-publish $(PUBLISH_DOCKER_ARCHS)
```

Comment:
Not much of a Makefile expert but I don't think `$(PUBLISH_DOCKER_ARCHS)` should be here, right? It should only mark the target itself as PHONY?

Reply:
All of those are actually targets, which could also be called individually.

E.g. common-docker-publish-AMD64

And therefor I think it makes sense to make the all and the sub targets phony